### PR TITLE
feat: database + server connection

### PR DESCRIPTION
**Pull request by JozefFlakus**
## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
- Separated app bootstrapping into dedicated files: `server.ts` and `database.ts`
- Added basic integration tests for `api.controller.ts`
- Specified which files should be included inside test coverage report

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,26 +44,30 @@
...
     "@marblejs/core": "^1.0.0-rc.2",
-    "@marblejs/middleware-joi": "^1.0.0-rc.2",
     "@marblejs/middleware-body": "^1.0.0-rc.2",
+    "@marblejs/middleware-joi": "^1.0.0-rc.2",
     "@marblejs/middleware-logger": "^1.0.0-rc.2",
+    "@types/chalk": "^2.2.0",
+    "@types/mongoose": "^5.2.12",
     "@types/node": "^10.1.2",
+    "chalk": "^2.4.1",
+    "mongoose": "^5.2.14",
```
Comment (krzysztof-miemiec) [inline package.json]: Woah woah woah what happens here? 😮 😃

File: src/connection/database.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  const onError = (error: mongoose.Error) => {
+    console.error(chalk.red(`[database] connection error: ${error.message}`));
+    process.exit();
+  };
+
+  export const connect = () =>
+    mongoose
+      .connect(url, { useNewUrlParser: true })
+      .then((onOpen))
```
Comment (krzysztof-miemiec) [inline src/connection/database.ts]: unnecessary `()`

File: src/index.ts [hunk 3]
```diff
@@ -1,12 +1,10 @@
-import { createServer } from 'http';
-import { Config } from './config';
+import { Database } from './connection/database';
+import { Server } from './connection/server';
 import { app } from './app';
 
-const SERVER_HOSTNAME = Config.server.host;
-const SERVER_PORT = Config.server.port;
+const bootstrap = async () => {
```
Comment (krzysztof-miemiec) [inline src/index.ts]: I never liked `bootstrap` name. Can we change it? 😅

Comment (JozefFlakus) [reply src/index.ts]: Nope, I like `bootstrap` 😄

Comment (JozefFlakus) [reply src/index.ts]: `main` is so Java

File: package.json (hunk 1 above)
Comment (JozefFlakus) [reply package.json]: What , what happened here!?

Comment (krzysztof-miemiec) [reply package.json]: Mongoose! It means that our example is getting mature and more advanced  👏

Comment (JozefFlakus) [reply package.json]: Yup, thats how we will do it. We will connect it with `mLab`

File: src/index.ts (hunk 3 above)
Comment (krzysztof-miemiec) [reply src/index.ts]: How about `run`? Or something that will remind everyone that Marble is about streams, like `flow` 😄

**Pull request merged**